'skaffold init' for Kustomize projects generates profiles for each overlay (#4349)
GoogleContainerTools/skaffold: docs/content/en/docs/pipeline-stages/init.md

    - leeroy-web/kubernetes/deployment.yaml
```

For projects deploying with `kustomize`, Skaffold will scan your project and look for `kustomization.yaml`s as well as Kubernetes manifests.
It will attempt to infer the project structure based on the recommended project structure from the Kustomize project: thus, 
**it is highly recommended to match your project structure to the recommended base/ and overlay/ structure from Kustomize!**

> This generally looks like this:

Reviewer: ```suggestion
This project structure generally looks like this:
```

```
app/      <- application source code, along with build configuration
  main.go
  Dockerfile
...
base/     <- base deploy configuration
  kustomization.yaml
  deployment.yaml
overlays/ <- one or more nested directories, each with modified environment configuration
  dev/
    deployment.yaml
    kustomization.yaml
  prod/
...
```

> When overlay directories are found, these will be listed in the generated Skaffold config as `paths` in the `kustomize` deploy stanza. However, it generally does not make sense to have multiple overlays applied at the same time, so **Skaffold will attempt to choose a default overlay, and put each other overlay into its own profile**. This can be specified by the user through the flag `--default-kustomization`; otherwise, Skaffold will use the following heuristic:

Reviewer: ```suggestion
When overlay directories are found, these will be listed in the generated Skaffold config as `paths` in the `kustomize` deploy stanza. However, it generally does not make sense to have multiple overlays applied at the same time, so **Skaffold will attempt to choose a default overlay, and put each other overlay into its own profile**. The default overlay can be specified by the user through the flag `--default-kustomization`; otherwise, Skaffold will use the following heuristic:
```

1) Any overlay with the name `dev`
2) If none present, the **first** overlay that isn't named `prod`

*Note: order is guaranteed, since Skaffold's directory parsing is always deterministic.*

## Init API

> `skaffold init` also exposes an AP:I which tools like IDEs can integrate with via flags.

Reviewer 2: ```suggestion
`skaffold init` also exposes an API which tools like IDEs can integrate with via flags.
```
